src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManager.cs

```diff
@@ -136,24 +136,24 @@ public override async Task<T> InvokeConnectionAsync<T>(string connectionId, stri
             {
                 var message = AppendMessageTracingId(new ClientInvocationMessage(invocationId, connectionId, _callerId, SerializeAllProtocols(methodName, args, invocationId)));
                 await WriteAsync(message);
+
+                string instanceId = null;
                 if (_clientConnectionManager.ClientConnections.TryGetValue(connectionId, out var clientConnectionContext))
                 {
-                    var instanceId = clientConnectionContext.InstanceId;
-                    var task = _clientInvocationManager.Caller.AddInvocation<T>(connectionId, invocationId, instanceId, cancellationToken);
-                    return await task;
+                    instanceId = clientConnectionContext.InstanceId;
                 }
+                var task = _clientInvocationManager.Caller.AddInvocation<T>(connectionId, invocationId, instanceId, cancellationToken);
+                return await task;
             }
-            catch
+            catch(Exception e)
             {
-                // Use `TryCompleteResult` to remove the failed invoation.
+                // Use `TryCompleteResult` to remove the failed invocation.
                 // If the invocation was not added by caller, it will be ignored.
                 // The content of error message is useless and will not be exposed to client.
-                _clientInvocationManager.Caller.TryCompleteResult(connectionId, CompletionMessage.WithError(invocationId, "Invocation Failed"));
+                _clientInvocationManager.Caller.TryCompleteResult(connectionId, CompletionMessage.WithError(invocationId, ""));
```

Comment:
Error message in `CompletionMessage.WithError` is not exposed to the customer? Then what is it for?